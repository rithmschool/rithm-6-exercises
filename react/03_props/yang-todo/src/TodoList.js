import React, { Component } from 'react';
import Todo from './Todo.js'
import NewTodoForm from "./NewTodoForm.js"

export default class TodoList extends Component {
    constructor(props) {
        super(props);
        this.state = {
            tasks: []
        }
        this.addTodo = this.addTodo.bind(this);
    };

    addTodo(event) {
        this.setState({ 
            tasks: this.state.tasks.concat([event.target.task.value])
          })
        event.preventDefault()
        event.target.reset()
        return;
    }

    render() {
        console.log(this.state.tasks);
        var taskList = this.state.tasks.map((task,idx) => <Todo value={task} key={idx}/>
        );
        return (
            <div>
                <NewTodoForm addTodo={this.addTodo} />
                {taskList}
            </div>
        )
    }
}