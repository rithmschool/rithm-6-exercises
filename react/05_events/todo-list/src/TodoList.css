.TodoList {
  display: flex;
  align-items: center;
  flex-direction: column;
  width: 100vw;
  height: 40rem;
  background-color: rgb(110, 226, 43);
}
