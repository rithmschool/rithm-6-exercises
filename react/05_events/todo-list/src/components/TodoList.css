.TodoList {
  display: flex;
  align-items: center;
  flex-direction: column;
  width: 100vw;
  height: 40rem;
}

img {
  height: 10rem;
  border-radius: 30px 30px 30px 30px;
}
