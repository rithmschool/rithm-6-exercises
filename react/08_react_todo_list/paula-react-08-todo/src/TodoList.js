import React, { Component } from "react";
import "./Todo.css";
import Todo from "./Todo";
import TodoForm from "./TodoForm";

export default class TodoList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      todos: [
        { title: "laundry", description: "do laundry", isComplete: false },
        {
          title: "grocery",
          description: "go grocery shopping",
          isComplete: false
        }
      ]
    };
    this.handleAdd = this.handleAdd.bind(this);
    this.toggleComplete = this.toggleComplete.bind(this);
  }
  handleAdd(newTodo) {
    newTodo.isComplete = false;
    this.setState(prevState => ({ todos: [newTodo, ...prevState.todos] }));
  }

  toggleComplete(idx) {
    let newTodos = [...this.state.todos];
    newTodos[idx].isComplete = !newTodos[idx].isComplete;
    this.setState({ todos: newTodos });
  }

  handleDelete(idx) {
    let updatedTodos = this.state.todos.filter((todo, i) => i !== idx);
    this.setState({ todos: updatedTodos });
  }

  handleEdit(idx, updatedTodo) {
    let updatedTodos = [...this.state.todos];
    updatedTodos[idx][updatedTodo.field] = updatedTodo.value;
    this.setState({ todos: updatedTodos });
  }

  render() {
    let todos = this.state.todos.map((t, i) => (
      <Todo
        key={i}
        title={t.title}
        description={t.description}
        isComplete={t.isComplete}
        toggleComplete={this.toggleComplete.bind(this, i)}
        handleDelete={this.handleDelete.bind(this, i)}
        handleEdit={this.handleEdit.bind(this, i)}
      />
    ));
    return (
      <section className="TodoList">
        <h2>Add a new todo!</h2>
        <TodoForm handleAdd={this.handleAdd} />
        <h3>Todos</h3>
        {todos}
      </section>
    );
  }
}
