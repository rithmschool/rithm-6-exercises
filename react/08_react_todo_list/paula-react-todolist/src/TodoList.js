import React, { Component } from "react";

export default class TodoList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      todos: []
    };
  }
}
