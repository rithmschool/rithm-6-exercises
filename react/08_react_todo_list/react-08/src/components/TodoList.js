import React, { Component } from 'react';
import Todo from './Todo';

class TodoList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      todos: [
        { id: 0, task: 'eat' },
        { id: 1, task: 'sleep' },
        { id: 2, task: 'wake up happy' },
        { id: 3, task: 'be grateful' },
        { id: 4, task: 'code' },
        { id: 5, task: 'kick ass' },
        { id: 6, task: 'chew bubblegum' }
      ]
    };
  }

  render() {
    var todos = this.state.todos.map(taco => <li> {taco.task} </li>);

    const someStyle = { color: 'green' };

    // var todos = this.state.todos.map(function(taco, idx) {
    //   return <li> {taco.task} </li>;
    // });

    return <div style={someStyle}> {todos} </div>;
  }
}

export default TodoList;
