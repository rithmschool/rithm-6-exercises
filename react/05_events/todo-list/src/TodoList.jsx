import React, { Component } from "react";
import Todo from "./Todo";
import "./TodoList.css";
import NewTodoForm from "./NewTodoForm";

class TodoList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      todos: ["test2"]
    };
    this.handleAdd = this.handleAdd.bind(this);
    this.handleDelete = this.handleDelete.bind(this);
    this.handleUpdate = this.handleUpdate.bind(this);
  }

  handleAdd(newTodo) {
    this.setState(prevState => ({
      todos: [newTodo, ...prevState.todos]
    }));
  }
  handleUpdate(newText, i) {
    let statei = i;
    this.setState(prevState => {
      this.state.todos[statei] = newText;
      return this.state;
    });
  }
  handleDelete(e) {
    let statei = e.target.getAttribute("index");
    this.setState(prevState => {
      this.state.todos.splice(statei, 1);
      return this.state;
    });
  }

  render() {
    let todos = this.state.todos.map((todo, i) => (
      <li>
        <Todo
          key={i}
          text={todo}
          index={i}
          handleDelete={this.handleDelete}
          handleUpdate={this.handleUpdate}
        />
      </li>
    ));
    return (
      <div className="TodoList">
        <NewTodoForm handleAdd={this.handleAdd} />
        <ul>{todos}</ul>
      </div>
    );
  }
}

export default TodoList;
