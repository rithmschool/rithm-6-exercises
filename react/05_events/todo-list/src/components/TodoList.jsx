import React, { Component } from "react";
import Todo from "./Todo";
import "./TodoList.css";
import { Route, Link } from "react-router-dom";
import NewTodoForm from "./NewTodoForm";

class TodoList extends Component {
  constructor(props) {
    super(props);
    this.handleUpdate = this.handleUpdate.bind(this);
  }

  handleUpdate(newText, i) {
    this.props.updateTodo(newText, i);
  }
  handleDelete(i) {
    this.props.removeTodo(i);
  }

  render() {
    let todos = this.props.todos.map((todo, i) => (
      <li>
        <Todo
          key={i}
          text={todo}
          index={i}
          handleDelete={this.handleDelete.bind(this, i)}
          handleUpdate={this.handleUpdate}
        />
      </li>
    ));
    return (
      <div className="TodoList">
        <Link to="/new">Add New Todo</Link>
        <ul>{todos}</ul>
        <img
          src="https://i.pinimg.com/originals/81/35/08/8135084de2993949cc18cb9f57f2aa48.png"
          alt=""
        />
      </div>
    );
  }
}

export default TodoList;
