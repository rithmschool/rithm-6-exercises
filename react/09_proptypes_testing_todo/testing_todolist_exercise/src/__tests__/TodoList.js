import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import TodoList from '../TodoList';

describe('<TodoList />', () => {

  let wrapper;

  it('renders', () => {
    wrapper = shallow(<TodoList />);
  });

  it('matches snapshot', () => {
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

});
