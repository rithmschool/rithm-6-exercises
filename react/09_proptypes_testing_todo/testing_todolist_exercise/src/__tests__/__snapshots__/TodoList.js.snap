// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> matches snapshot 1`] = `
<div>
  <NewTodoForm
    handleAdd={[Function]}
  />
  <ol>
    <Todo
      description="Grocery Store"
      handleComplete={[Function]}
      handleDelete={[Function]}
      handleEdit={[Function]}
      handleEditRequest={[Function]}
      isEditing={false}
      title="Buy ice cream"
    />
    <Todo
      description="Dinner Plans"
      handleComplete={[Function]}
      handleDelete={[Function]}
      handleEdit={[Function]}
      handleEditRequest={[Function]}
      isEditing={false}
      title="Order pizza"
    />
    <Todo
      description="Fitness Routine"
      handleComplete={[Function]}
      handleDelete={[Function]}
      handleEdit={[Function]}
      handleEditRequest={[Function]}
      isEditing={false}
      title="Walk 20 miles"
    />
  </ol>
</div>
`;
